Use Font Awesome SVG icons to indicate file status

marlo-longley opened the pull request:
Fixes #425 

There were many ways to approach changing the icons (css only vs. markup output vs. rails helpers) but this approach avoided the need to import a font awesome or bootstrap icons library, and stayed in line with the existing logic. 

I also wanted to test for the fact the the correct status classes were being applied - there may be better places to test as well. 

Here's how it's looking:
![Screen Shot 2022-03-24 at 14 20 31](https://user-images.githubusercontent.com/1328900/160017659-b61458f4-848d-49d1-b930-0efc69e165f4.png)

(unknown status:)
![Screen Shot 2022-03-24 at 14 25 10](https://user-images.githubusercontent.com/1328900/160017673-e16a4de0-fb4a-4e5c-b1d1-7a4035cd12d4.png)

Other styling issues such as vertical alignment within the table is outside the scope of this issue and can be fixed when we work on tables in the larger redesign.

corylown commented on app/helpers/bootstrap_icons_helper.rb:
```diff
@@ -0,0 +1,15 @@
+# Helper to render Bootstrap Icons
+module BootstrapIconsHelper
+  # refer https://icons.getbootstrap.com/
+  def icon(icon, options = {})
```
I think I'd move this method to the existing `ApplicationHelper` and maybe name the method `bootstrap_icon` for clarity. Maybe add a simple test too?

corylown commented on app/helpers/bootstrap_icons_helper.rb:
```diff
@@ -0,0 +1,15 @@
+  # refer https://icons.getbootstrap.com/
+  def icon(icon, options = {})
+    file = File.read("node_modules/bootstrap-icons/icons/#{icon}.svg")
+    content_tag(:span, class: options[:class].presence) do
```
As far as I can tell `options[:class]` and `options[:class].presence` both just return nil if there's not a `:class` key so I don't think you need to call `.presence`

corylown commented on config/settings.yml:
```diff
@@ -19,3 +19,11 @@ marc_fixture_seeds:
     - Stanford
+
+# refer to https://icons.getbootstrap.com/ for icon class names
+metadata_status_icons:
```
I'm wondering why this is in settings? It doesn't seem like something that we're likely to configure differently per environment. Maybe a helper method would work?

in `ApplicationHelper`:
```ruby
def metadata_status_icons(status)
  case status
  when :deletes then 'file-earmark-excel-fill'
  when :invalid then 'exclamation-circle-fill'
  # etc...
end
```

corylown reviewed (changes requested):
A few inline questions/comments. The original issue shows a legend above the table explaining the meaning of the icons, which I think would be nice to add to this PR: https://github.com/pod4lib/aggregator/issues/425

marlo-longley commented:
@corylown thanks for your careful review, I will address.

marlo-longley commented:
@corylown I've addressed your comments, thanks for your help!
You can see the legend looks like this now:

<img width="1353" alt="Screen Shot 2022-03-30 at 17 42 18" src="https://user-images.githubusercontent.com/1328900/160936031-6155b8f9-0a91-41a3-9595-02790720f5d5.png">
